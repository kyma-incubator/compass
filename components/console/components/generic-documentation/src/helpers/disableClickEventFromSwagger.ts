export function disableClickEventFromSwagger() {
  const beginOfSwaggerClickFunctionAsString = `for (var t = e.target; t && "A" !== t.tagName;)`;

  const originalEventListener = window.addEventListener;
  window.addEventListener = (
    type: string,
    fn: (...args: any[]) => any,
    options: any,
  ) => {
    if (type === 'click') {
      const fnAsString = fn.toString();
      if (fnAsString.includes(beginOfSwaggerClickFunctionAsString)) {
        return;
      }
    }
    return originalEventListener(type, fn, options);
  };
}
