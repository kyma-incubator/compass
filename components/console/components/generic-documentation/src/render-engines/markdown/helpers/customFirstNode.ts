import { Source } from '@kyma-project/documentation-component';
import { plugins } from '@kyma-project/dc-markdown-render-engine';
import { headingPrefix } from './headingPrefix';

type Header = plugins.Header;

export function customFirstNode(
  source: Source,
  toKebabCase: (str: string) => string,
): Header | undefined {
  const title: string =
    source.data && source.data.frontmatter && source.data.frontmatter.title;

  if (!title) {
    return;
  }

  return {
    title,
    id: toKebabCase(headingPrefix(source)),
    level: 'doc-title',
    source,
  };
}
