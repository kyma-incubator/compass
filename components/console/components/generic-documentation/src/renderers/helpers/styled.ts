import styled from 'styled-components';
import { ComboboxInput, Menu } from 'fundamental-react';

const typeColumnWidth = '6em';
const listItemColumnGap = '1em';
const listItemPadding = '2 * 10px';
export const ListItem = styled.span`
   {
    justify-items: start;
    display: grid;
    grid-gap: ${listItemColumnGap};
    grid-template-columns: ${typeColumnWidth} auto;
  }
`;

export const ApiTabHeader = styled.div`
   {
    display: grid;
    align-items: center;
    grid-template-columns: auto auto;
    grid-gap: 1em;
  }
`;

export const Combobox = styled(ComboboxInput)`
   {
    flex-shrink: 0;
    min-width: calc(
      ${props => props['data-max-list-chars'] + 'ch'} + ${typeColumnWidth} +
        ${listItemColumnGap} + ${listItemPadding}
    );
  }
`;

export const List = styled(Menu)`
   {
    max-height: 28em;
    overflow-y: auto;
  }
`;
