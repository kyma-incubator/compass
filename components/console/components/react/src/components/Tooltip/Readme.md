```jsx
<div>
  <Tooltip title="info" content="Content inside tooltip." minWidth="160px" color="#000" type="error">
    Content for tooltip
  </Tooltip>
</div>
```
