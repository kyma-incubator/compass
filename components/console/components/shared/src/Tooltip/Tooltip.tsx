import React, { useState } from 'react';

import {
  TooltipWrapper,
  TooltipContainer,
  TooltipContent,
  TooltipHeader,
} from './styled';

export enum TooltipType {
  STANDARD = '',
  LIGHT = 'light',
  INFO = 'info',
  POSITIVE = 'positive',
  WARNING = 'warning',
  NEGATIVE = 'negative',
}

export enum TooltipOrientation {
  TOP = 'top',
  BOTTOM = 'bottom',
}

export interface TooltipProps {
  title?: React.ReactNode;
  content: React.ReactNode;
  minWidth?: string;
  maxWidth?: string;
  type?: TooltipType;
  show?: boolean;
  showTooltipTimeout?: number;
  hideTooltipTimeout?: number;
  orientation?: TooltipOrientation;
  wrapperStyles?: string;
}

export const Tooltip: React.FunctionComponent<TooltipProps> = ({
  title,
  content,
  minWidth,
  maxWidth,
  type = TooltipType.STANDARD,
  show = false,
  showTooltipTimeout = 100,
  hideTooltipTimeout = 100,
  orientation = TooltipOrientation.TOP,
  wrapperStyles,
  children,
}) => {
  const [visibleTooltip, setVisibleTooltip] = useState<boolean>(false);

  const handleShowTooltip = () => {
    if (!show) {
      setTimeout(() => setVisibleTooltip(true), showTooltipTimeout);
    }
  };

  const handleHideTooltip = () => {
    if (!show) {
      setTimeout(() => setVisibleTooltip(false), hideTooltipTimeout);
    }
  };

  return (
    <TooltipWrapper
      onMouseEnter={handleShowTooltip}
      onMouseLeave={handleHideTooltip}
      type={type}
      wrapperStyles={wrapperStyles}
    >
      {children}
      {visibleTooltip && content && (
        <TooltipContainer
          minWidth={minWidth}
          maxWidth={maxWidth}
          type={type}
          show={show}
          orientation={orientation}
        >
          {title && <TooltipHeader type={type}>{title}</TooltipHeader>}
          <TooltipContent type={type}>{content}</TooltipContent>
        </TooltipContainer>
      )}
    </TooltipWrapper>
  );
};
