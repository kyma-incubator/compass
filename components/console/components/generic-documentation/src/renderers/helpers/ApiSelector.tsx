import React, { useState } from 'react';
import { Combobox, List, ListItem } from './styled';
import { Badge } from 'fundamental-react';
import {
  odataDefinition,
  asyncApiDefinition,
  markdownDefinition,
} from '../../constants';
import { Source } from '@kyma-project/documentation-component';

function getApiNameLength(s: Source) {
  return s.data && s.data.displayName ? s.data.displayName.length : 0;
}

const BadgeForType: React.FunctionComponent<{ type: string }> = ({ type }) => {
  let badgeType: 'success' | 'warning' | 'error' | undefined;

  if (odataDefinition.possibleTypes.includes(type)) {
    badgeType = 'warning';
  }

  if (asyncApiDefinition.possibleTypes.includes(type)) {
    badgeType = 'success';
  }

  if (markdownDefinition.possibleTypes.includes(type)) {
    badgeType = 'error';
  }

  return <Badge type={badgeType}>{type}</Badge>;
};

const ApiSelector: React.FunctionComponent<{
  sources: Source[];
  onApiSelect: (api: Source) => void;
  selectedApi: Source;
}> = ({ sources, onApiSelect, selectedApi }) => {
  const [searchText, setSearchText] = useState('');

  const filteredSources = sources.filter(
    (s: Source) =>
      (s.data &&
        s.data.displayName &&
        s.data.displayName.toUpperCase().includes(searchText.toUpperCase())) ||
      s.type.includes(searchText),
  );

  function handleInputChange(e: React.ChangeEvent<HTMLInputElement>) {
    setSearchText(e.target.value);
  }

  const maxApiNameLength = filteredSources.length
    ? getApiNameLength(
        filteredSources.reduce((prev, current) =>
          getApiNameLength(current) > getApiNameLength(prev) ? current : prev,
        ),
      )
    : 0;

  return (
    <Combobox
      onClick={(e: React.MouseEvent<HTMLElement>) => {
        const a = e.target as HTMLElement; // not sure why but it's needed, thank you typescript!
        if (a.tagName === 'INPUT' || a.tagName === 'BUTTON') {
          e.stopPropagation(); // avoid closing the dropdown due to the "opening" click âˆž
        }
      }}
      data-max-list-chars={maxApiNameLength}
      menu={
        <List>
          {filteredSources.map((s: Source, id) => (
            <a
              aria-label="api-"
              href="#"
              onClick={e => onApiSelect(s)}
              className="fd-menu__item"
              key={(s.data && s.data.displayName) || s.rawContent}
            >
              <ListItem>
                <BadgeForType type={s.type} />
                {s.data && s.data.displayName}
              </ListItem>
            </a>
          ))}
        </List>
      }
      placeholder={
        (selectedApi && selectedApi.data && selectedApi.data.displayName) ||
        (selectedApi && selectedApi.type) ||
        'Select API'
      }
      inputProps={{ onChange: handleInputChange }}
    />
  );
};

export default ApiSelector;
