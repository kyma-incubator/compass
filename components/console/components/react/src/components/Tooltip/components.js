import styled from 'styled-components';

export const TooltipContainer = styled.div`
  position: absolute;
  box-sizing: border-box;
  z-index: 99;
  min-width: ${props => (props.minWidth ? props.minWidth : '120px')};
  max-width: ${props => (props.maxWidth ? props.maxWidth : '420px')};
  background: ${props =>
    props.type === 'default' && !props.light ? '#32363a' : '#fff'};
  font-size: ${props => (props.type === 'default' ? '11px' : '12px')};
  line-height: ${props => (props.type === 'default' ? '11px' : '12px')};
  color: ${props =>
    props.type === 'default' && !props.light ? '#fff' : '#32363b'};
  filter: drop-shadow(rgba(0, 0, 0, 0.12) 0 0px 2px);
  box-shadow: 0 0 4px 0 #00000026, 0 12px 20px 0 #00000019;
  border-radius: 3px;
  border-left: ${props => {
    let color = '';
    switch (props.type) {
      case 'info':
        color = '#0b74de';
        break;
      case 'warning':
        color = '#ffeb3b';
        break;
      case 'success':
        color = '#4caf50';
        break;
      case 'error':
        color = '#f44336';
        break;
      default:
        color = 'none';
    }
    return `6px solid ${color}`;
  }};
  ${props => (props.orientation === 'top' ? 'bottom: 100%;' : 'top: 100%')};
  right: 50%;

  ${props => (props.type === 'light' && 'left: 0;') || 'right: 50%'};
  transform: translateX(
    ${props =>
      (props.type === 'default' && '50%') ||
      (props.type === 'light' && '-40px') ||
      '40px'}
  );
  visibility: ${props => (props.show ? 'visibility' : 'hidden')};
  opacity: ${props => (props.show ? '1' : '0')};
  ${props => {
    switch (props.orientation) {
      case 'bottom':
        return `margin-top: ${props.type === 'default' ? '8px' : '16px'}`;
      default:
        return `margin-bottom: ${props.type === 'default' ? '8px' : '16px'}`;
    }
  }};

  &:after {
    border: ${props => (props.type === 'default' ? '6px' : '10px')} solid;
    border-color: ${props => {
      switch (props.orientation) {
        case 'bottom':
          return `transparent transparent ${
            props.type === 'default' ? '#32363b' : '#fff'
          }`;
        default:
          return `${
            props.type === 'default' ? '#32363b' : '#fff'
          } transparent transparent`;
      }
    }};
    content: '';
    ${props =>
      (props.type === 'default' && 'right: 50%;') ||
      (props.type === 'light' && 'left: 48px;') ||
      'right: 25px'};
    ${props => (props.type === 'default' ? 'transform: translateX(6px)' : '')};
    margin-left: -10px;
    position: absolute;
    ${props =>
      props.orientation === 'top'
        ? 'top: 100%; margin-top: -1px;'
        : 'bottom: 100%; margin-bottom: -1px;'};
  }
`;

export const TooltipWrapper = styled.div`
  font-family: '72';
  position: relative;
  display: inline-block;

  ${props => (props.wrapperStyles ? props.wrapperStyles : '')}

  &:hover ${TooltipContainer} {
    visibility: visible;
    opacity: 1;
  }
`;

export const TooltipContent = styled.div`
  display: block;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: ${props => (props.type === 'default' ? 'center' : 'left')};
  padding: ${props => (props.type === 'default' ? '6px 10px' : '12px 14px')};
`;

export const TooltipHeader = styled.div`
  border-bottom: 1px solid rgba(204, 204, 204, 0.3);
  font-weight: bold;
  text-align: left;
  position: relative;
  padding: 12px 14px;
  box-sizing: border-box;

  &:after {
    ${props => {
      switch (props.type) {
        case 'info':
          return "content: '\uE1C3'; color: #0b74de;";
        case 'warning':
          return "content: '\uE053'; color: #ffeb3b;";
        case 'success':
          return "content: '\uE1C1'; color: #4caf50;";
        case 'error':
          return "content: '\uE0B1'; color: #f44336;";
        default:
          return "content: '';";
      }
    }};
    position: absolute;
    display: block;
    top: 12px;
    right: 14px;
    box-sizing: border-box;
    font-family: SAP-Icons;
  }
`;

export const TooltipFooter = styled.div`
  border-bottom: 1px solid #ccc;
  line-height: 40px;
  font-family: 72;
  font-weight: bold;
  text-align: left;
  position: relative;
  padding: 0 12px;
  box-sizing: border-box;
`;
