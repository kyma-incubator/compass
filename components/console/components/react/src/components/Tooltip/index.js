import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Icon from '../Icon';

import {
  TooltipWrapper,
  TooltipContainer,
  TooltipContent,
  TooltipHeader,
  TooltipFooter,
} from './components';

class Tooltip extends React.Component {
  static propTypes = {
    children: PropTypes.any.isRequired,
    content: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array,
      PropTypes.element,
    ]),
    show: PropTypes.bool,
    title: PropTypes.string,
    type: PropTypes.string,
    minWidth: PropTypes.string,
    maxWidth: PropTypes.string,
    orientation: PropTypes.string,
    showTooltipTimeout: PropTypes.number,
    hideTooltipTimeout: PropTypes.number,
  };

  static defaultProps = {
    orientation: 'top',
  };

  constructor(props) {
    super(props);
    this.state = {
      visibleTooltip: this.props.show === undefined ? false : this.props.show,
      showTooltip: this.props.show === undefined ? false : this.props.show,
    };
  }

  setVisibility = visible => {
    this.setState({ visibleTooltip: visible });
  };

  handleShowTooltip = () => {
    const { showTooltipTimeout } = this.props;
    if (typeof this.setVisibility === 'function' && !this.state.showTooltip) {
      setTimeout(
        () => this.setVisibility(true),
        showTooltipTimeout ? showTooltipTimeout : 100,
      );
    }
  };

  handleHideTooltip = () => {
    const { hideTooltipTimeout } = this.props;
    if (typeof this.setVisibility === 'function' && !this.state.showTooltip) {
      setTimeout(
        () => this.setVisibility(false),
        hideTooltipTimeout ? hideTooltipTimeout : 100,
      );
    }
  };

  render() {
    const { visibleTooltip, showTooltip } = this.state;
    const {
      children,
      title,
      content,
      minWidth,
      maxWidth,
      icon,
      type,
      orientation,
      wrapperStyles,
    } = this.props;

    return (
      <TooltipWrapper
        onMouseEnter={this.handleShowTooltip}
        onMouseLeave={this.handleHideTooltip}
        type={type === undefined ? 'default' : type}
        wrapperStyles={wrapperStyles}
      >
        {children}
        {visibleTooltip && content && (
          <TooltipContainer
            minWidth={minWidth}
            maxWidth={maxWidth}
            type={type === undefined ? 'default' : type}
            show={showTooltip}
            orientation={orientation}
          >
            {title && (
              <TooltipHeader type={type === undefined ? 'default' : type}>
                {title}
              </TooltipHeader>
            )}
            <TooltipContent type={type === undefined ? 'default' : type}>
              {content}
            </TooltipContent>
          </TooltipContainer>
        )}
      </TooltipWrapper>
    );
  }
}

export default Tooltip;
