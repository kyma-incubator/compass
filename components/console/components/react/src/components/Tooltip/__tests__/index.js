import React from 'react';
import Tooltip from '../index';

import { renderWithTheme } from 'test_setup/helpers';

describe('Tooltip Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Tooltip content="Content inside">Content for tooltip</Tooltip>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
