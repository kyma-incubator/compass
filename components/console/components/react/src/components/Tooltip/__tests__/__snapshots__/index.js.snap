// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Element renders correctly 1`] = `
<div
  className="sc-htpNat gHmJEJ"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="default"
>
  Content for tooltip
</div>
`;
